### PR TITLE
ENGA-1120 Add twothirdblast type to email metrics reporting

### PR DESCRIPTION
**Pull request by ddicocco**
[ENGA-1120 Add twothirdblast type to email metrics reporting](https://industrydive.atlassian.net/browse/ENGA-1120)

### Non-technical Context
This change updates the logic used by the Dive Sailthru Client in determining the dive_email_type in order to support new blast naming standards for new ad twothirdblast s.

### Technical Context
- Updates `_infer_dive_email_type()` to identify twothirdblasts by either campaign name or sailthru list name
  - The naming convention will be "___ Dive TWO Third Blast List – Group X", but kept convention of also looking for "...two third**s** blast list..." just in case
- Updates `_infer_dive_publication()` to accomodate new twothirdsblast
- Added test configs to confirm added logic

#### Developer

- [X] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
- [X] Devise edge cases to test the bug fix or feature being merged
- [X] Make sure the code is clean and understandable
- [X] Ensure code is not overly inefficient
- [X] Ensure documentation is understandable and accurate, including:
    - [X] Code comments and doc strings
    - [N/A] Technical documentation
- [ ] Set deployment notes on Jira ticket

#### Reviewer

- [x] Ensure code complies with the [style guide](https://industrydive.atlassian.net/wiki/spaces/TECH/pages/639500407/Tech+Team+s+Style+Guides)
    - If there are few or minor issues, add inline GitHub comments
    - If there are many issues or it seems like the developer did not follow the style guide, leave a comment asking them to do so
- [x] Devise edge cases to test the bug fix or feature being merged
- [x] Make sure the code is clean and understandable
    - [ ] Ask for inline comments on unclear sections
- [x] Ensure code is not overly inefficient
- [x] Ensure documentation is understandable and accurate, including:
    - [ ] Code comments and doc strings
    - [ ] Technical documentation
- [x] Read any deployment notes on the Jira ticket and ensure they can be followed

### CODE REVIEW

File: dive_sailthru_client/client.py [hunk 1]
```diff
@@ -129,8 +133,9 @@ def _infer_dive_publication(self, campaign):
             # blast list: "Utility Dive and sub pubs Blast List"
             # This regex handles that as well as normal blast lists
             return re.sub(r'( and sub pubs)? [Bb]last [Ll]ist$', '', list_name)
-        if dive_email_type in (DiveEmailTypes.HalfBlast, DiveEmailTypes.QuarterBlast, DiveEmailTypes.ThirdBlast) and "Group" in list_name:
-            return re.sub(r'( (Half|Thirds?|Quarters?))? Blast List - Group [A-D]$', '', list_name, flags=re.I)
+        if (dive_email_type in (DiveEmailTypes.HalfBlast, DiveEmailTypes.QuarterBlast, DiveEmailTypes.ThirdBlast, DiveEmailTypes.TwoThirdBlast) and
+            "Group" in list_name):
+            return re.sub(r'( (Half|(TWO )?Thirds?|Quarters?))? Blast List - Group [A-D]$', '', list_name, flags=re.I)
```
Comment (DevDaveFrame) [inline dive_sailthru_client/client.py:138]: Just for my own edification, is there some stakeholder-y reason for the all caps TWO?

**Comment by DevDaveFrame** [comment]
@ddicocco LGTM!

File: dive_sailthru_client/client.py (hunk 1 above)
Comment (ddicocco) [reply dive_sailthru_client/client.py:138]: 👌 + w/ the `re.I` flag it actually doesn't matter, just included because the rest was included in its expected case even tho it's case insensitive
